include/vast/Dialect/HighLevel/HighLevelOps.td

```diff
@@ -1260,6 +1260,58 @@ class AlignOfExprOpBase< string mnemonic, list< Trait > traits = [] >
 def AlignOfExprOp          : AlignOfExprOpBase< "alignof.expr" >;
 def PreferredAlignOfExprOp : AlignOfExprOpBase< "preferred_alignof.expr" >;
 
+def OffsetOfNodeAttr
+  : HighLevel_Attr< "OffsetOfNode", "offset_of_node" > {
+    let parameters = (ins
+      "std::variant< unsigned int, ::mlir::StringAttr >":$value
```

Comment:
use specifically sized type, should it be `uint32_t` or `uint64_t`?

Reply:
The method to get the value form the LLVM `OffsetOfNode` returns just `unsigned`, so I copied it from there. It extracts the data by doing a bit shift on a `uintptr_t`. So… It depends?